src/components/FloatingLabelInput/FloatingLabelInput.component.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+
+export default class FloatingLabelInput extends Component {
+    state = {
+        name: this.props.name,
+        type: this.props.type,
+        children: this.props.children,
+        value: this.props.value,
+        active: false,
+    };
+
+    componentDidMount() {
+        if (this.state.value !== '' && this.state.value) {
+            this.setState({ active: true });
+        }
+    }
+
+    handleActivation = (e) => {
+        this.setState({ active: !!e.target.value, value: e.target.value });
+    };
+    render() {
+        const { name, type, children, value, active } = this.state;
+        return (
+            <div className="relative mb-2 bg-transparent">
+                <input
+                    className={[
+                        'w-full bg-transparent border border-t-0 border-r-0 border-l-0 border-black rounded-none outline-none  text-sm transition-all duration-200 ease-in-out p-2 text-black dark:text-white',
```

Comment:
not a huge fan of having that amount of classes, there are much cleaner styling strategies for react, just keep that in mind for your next projects (: